adoptingerlang.org

```diff
@@ -3405,27 +3542,269 @@ $ kubectl apply -k deployment/overlays/dev
+
+The target is =dev_release= because we want the ability to do live updates simply by recompiling and restarting inside the running image:
+
+#+BEGIN_SRC dockerfile
+# image to use in tilt when running the release
+FROM builder as dev_release
+
+COPY . .
+RUN rebar3 as tilt release
+
+ENTRYPOINT ["/app/_build/tilt/rel/service_discovery/bin/service_discovery"]
+CMD ["foreground"]
+#+END_SRC
+
+The =live_update= instructions will run in the container any time a file changes. Since a =dev_mode= release is used, the Rebar3 profile =tilt= keeps the default =dev_mode= of =true=, the update commands simply need to sync the =apps= directory to the running container, run =compile= and restart the release.
+
+Lastly in the =Tiltfile= the Kubernetes resources to deploy are configured and a watch is set to rerun the =Tiltfile= if any of the kustomize files changes.
+
+#+BEGIN_SRC python
+k8s_yaml(kustomize('deployment/overlays/dev'))
+
+watch_file('deployment/')
+#+END_SRC
+
+Tilt has built in support for kustomize so we use =kustomize('deployment/overlays/dev')= to render the =dev= overlay and pass to =k8s_yaml= which tells Tilt what Kubernetes resources to deploy and track.
+
+The important differences from the =base= overlay in the =dev= overlay's =kustomization.yaml= is the inclusion of the Postgres kustomize resources and a patch that is merged on the =base= resources:
+
+#+BEGIN_SRC yaml
+bases:
+- ../../base
+- ../../postgres
+patchesStrategicMerge:
+- flyway_job_patch.yaml
+#+END_SRC
+
+The =fly_job_patch.yaml= is used to configure the Flyway job to work with the Tilt setup:
+
+#+BEGIN_SRC yaml
+# For tilt we make an image named service_discovery_sql with the migrations.
+# This patch replaces the image used in the flyway migration job to match.
+apiVersion: batch/v1
+kind: Job
+metadata:
+  labels:
+    service: flyway
+  name: flyway
+spec:
+  template:
+    spec:
+      initContainers:
+      - name: service-discovery-sql
+        image: service_discovery_sql
+        volumeMounts:
+          - name: migrations
+            mountPath: /flyway/sql
+        command: ["/bin/sh"]
+        args: ["-c", "cp /app/sql/* /flyway/sql"]
+#+END_SRC
+
+This patch changes the image name in the =Job= resource to =service_discovery_sql=, the same name as the first =custom_build= image in the =Tiltfile=. Tilt will update the image with the latest tag, which it sets as =$EXPECTED_REF= in the environment of the Docker build command in =custom_build=, and rerun the =Job=. This way while =service_discovery= is running in the local Kubernetes if a new migration is added it will automatically be picked up and run on the database, keeping our development cluster in sync with our local development.
```

Comment:
```suggestion
This patch changes the image name in the =Job= resource to =service_discovery_sql=, the same name as the first =custom_build= image in the =Tiltfile=. Tilt will update the image with the latest tag, which it sets as =$EXPECTED_REF= in the environment of the Docker build command in =custom_build=, and reruns the =Job=. This way while =service_discovery= is running in the local Kubernetes if a new migration is added it will automatically be picked up and run on the database, keeping our development cluster in sync with our local development.
```